modules/rostests/apitests/user32/ScrollBarRedraw.c

```diff
@@ -0,0 +1,722 @@
+    RGB(0x00, 0x00, 0x80),  /* Dark Blue */
+    RGB(0x00, 0x80, 0x00),  /* Dark Green */
+    RGB(0x00, 0x80, 0x80),  /* Dark Cyan */
+    RGB(0x80, 0x00, 0x00),  /* Dark Red */
+    RGB(0x80, 0x00, 0x80),  /* Dark Magenta */
+    RGB(0x80, 0x80, 0x00),  /* Dark Yellow */
+    RGB(0xC0, 0xC0, 0xC0),  /* Light Gray */
+    RGB(0x80, 0x80, 0x80),  /* Dark Gray */
+    RGB(0x00, 0x00, 0xFF),  /* Blue */
+    RGB(0x00, 0xFF, 0x00),  /* Green */
+    RGB(0x00, 0xFF, 0xFF),  /* Cyan */
+    RGB(0xFF, 0x00, 0x00),  /* Red */
+    RGB(0xFF, 0x00, 0xFF),  /* Magenta */
+    RGB(0xFF, 0xFF, 0x00),  /* Yellow */
+    RGB(0xFF, 0xFF, 0xFF)   /* White */
+};
+static HBRUSH ColorBrushes[TEST_COLOR_COUNT] = { 0 };
+
+static BOOL HaveHRedraw = FALSE;
+static BOOL HaveVRedraw = FALSE;
+
+typedef enum _FSM_STATE
+{
+    FSM_STATE_START,
+    FSM_STATE_VSCR_SHOWN,
+    FSM_STATE_VSCR_HIDDEN,
+    FSM_STATE_HSCR_SHOWN,
+    FSM_STATE_HSCR_HIDDEN,
+    FSM_STATE_BSCR_SHOWN,
+    FSM_STATE_BSCR_HIDDEN,
+    FSM_STATE_WIDTH_SHRUNK,
+    FSM_STATE_WIDTH_EXPANDED,
+    FSM_STATE_HEIGHT_SHRUNK,
+    FSM_STATE_HEIGHT_EXPANDED,
+    FSM_STATE_BOTH_SHRUNK,
+    FSM_STATE_BOTH_EXPANDED,
+    FSM_STATE_END
+} FSM_STATE;
+
+#define FSM_STEP_PERIOD_MS 250
+
+static UINT_PTR FsmTimer = 0;
+static UINT CurrentColor = 0;
+static FSM_STATE FsmState = FSM_STATE_START;
+
+static int ClientWidth = 0;
+static int ClientHeight = 0;
+
+static int OrigWidth = 0;
+static int OrigHeight = 0;
+static int SmallWidth = 0;
+static int SmallHeight = 0;
+
+static void ColorsCleanup(void)
+{
+    UINT Iter;
+
+    for (Iter = 0; Iter < _countof(ColorBrushes); Iter++)
+    {
+        if (ColorBrushes[Iter] != NULL)
+        {
+            DeleteObject(ColorBrushes[Iter]);
+            ColorBrushes[Iter] = NULL;
+        }
+    }
+}
+
+static BOOL ColorsInit(void)
+{
+    UINT Iter;
+
+    assert(_countof(Colors) == _countof(ColorBrushes));
+
+    for (Iter = 0; Iter < _countof(ColorBrushes); Iter++)
+    {
+        ColorBrushes[Iter] = CreateSolidBrush(Colors[Iter]);
+        if (ColorBrushes[Iter] == NULL)
+        {
+            ColorsCleanup();
+            return FALSE;
+        }
+    }
+
+    return TRUE;
+}
+
+static void RunTestWindow(PCWSTR ClassName, PCWSTR WindowTitle, UINT ClassStyle)
+{
+    WNDCLASSW Class = { 0 };
+    HWND Window;
+    MSG  Message;
+    HINSTANCE hInst;
+
+    CurrentColor = 0;
+    hInst = GetModuleHandleW(NULL);
+
+    Class.style         = ClassStyle;
+    Class.lpfnWndProc   = WindowProc;
+    Class.cbClsExtra    = 0;
+    Class.cbWndExtra    = 0;
+    Class.hInstance     = hInst;
+    Class.hIcon         = LoadIcon(NULL, IDI_APPLICATION);
+    Class.hCursor       = LoadCursor(NULL, IDC_ARROW);
+    Class.hbrBackground = ColorBrushes[CurrentColor];
+    Class.lpszMenuName  = NULL;
+    Class.lpszClassName = ClassName;
+
+    if (!RegisterClassW(&Class))
+    {
+        skip("Failed to register window class \"%ls\", code: %ld\n",
```

Comment:
**Optional**: Personally I find code like that to become shorter and more readable when `\"` is replaced by just `'` because that doesn't need to be escaped. But we are really at nitpicking level already, I don't want to withheld my approval for things like that. Especially not when it is about test-code and not production code. In case you want to improve on that, there are several hits of that same style.
```suggestion
        skip("Failed to register window class '%ls', code: %ld\n",
```

Reply:
Alrighty, I'll switch it out.  The reason why I tend to go for double quotes is that "they represent strings in C", and the apostrophe is a common diacritics mark in English text, so it might be jarring to have some text with apostrophes enclosed in single-quotes.